x-pack/plugins/security_solution/cypress/e2e/detection_rules/persistent_rules_table_state.cy.ts

```diff
@@ -0,0 +1,631 @@
+          cy.reload();
+
+          expectRulesManagementTab();
+          expectRulesTableState();
+          expectTablePage(2);
+          expectManagementTableRules(['rule 6']);
+        });
+      });
+
+      it('persists after corrupting the url param data', () => {
+        visit(SECURITY_DETECTIONS_RULES_MANAGEMENT_URL, { qs: { rulesTable: '(!invalid)' } });
+
+        expectRulesManagementTab();
+        expectRulesTableState();
+        expectTablePage(1);
+        expectManagementTableRules(['rule 1', 'rule 2', 'rule 3', 'rule 4', 'rule 5']);
+      });
+
+      it('DOES NOT persist after corrupting the session storage and url param data', () => {
+        visit(SECURITY_DETECTIONS_RULES_MANAGEMENT_URL, {
+          qs: { rulesTable: '(!invalid)' },
+          onBeforeLoad: (win) => {
+            win.sessionStorage.setItem('securitySolution.rulesTable', '!invalid');
+          },
+        });
+
+        expectRulesManagementTab();
+        expectDefaultRulesTableState();
+        expectManagementTableRules([
+          'test 1',
+          'rule 1',
+          'rule 2',
+          'rule 3',
+          'rule 4',
+          'rule 5',
+          'rule 6',
+          'rule 7',
+        ]);
+      });
+    });
+
+    describe('and on the rules monitoring tab', () => {
+      beforeEach(() => {
+        visit(SECURITY_DETECTIONS_RULES_MONITORING_URL);
+
+        changeRulesTableState();
+        goToTablePage(2);
+      });
+
+      it('persists after corrupting the session storage data', () => {
+        cy.window().then((win) => {
+          win.sessionStorage.setItem('securitySolution.rulesTable', '!invalid');
+          cy.reload();
+
+          expectRulesMonitoringTab();
+          expectRulesTableState();
+          expectTablePage(2);
+          expectMonitoringTableRules(['rule 6']);
+        });
+      });
+
+      it('persists after corrupting the url param data', () => {
+        visit(SECURITY_DETECTIONS_RULES_MONITORING_URL, { qs: { rulesTable: '(!invalid)' } });
+
+        expectRulesMonitoringTab();
+        expectRulesTableState();
+        expectTablePage(1);
+        expectMonitoringTableRules(['rule 1', 'rule 2', 'rule 3', 'rule 4', 'rule 5']);
+      });
+
+      it('DOES NOT persist after corrupting the session storage and url param data', () => {
+        visit(SECURITY_DETECTIONS_RULES_MONITORING_URL, {
+          qs: { rulesTable: '(!invalid)' },
+          onBeforeLoad: (win) => {
+            win.sessionStorage.setItem('securitySolution.rulesTable', '!invalid');
+          },
+        });
+
+        expectRulesMonitoringTab();
+        expectDefaultRulesTableState();
+        expectMonitoringTableRules([
+          'test 1',
+          'rule 1',
+          'rule 2',
+          'rule 3',
+          'rule 4',
+          'rule 5',
+          'rule 6',
+          'rule 7',
+        ]);
+      });
+    });
+  });
+
+  describe('in a new browser tab', () => {
```

Comment:
I also wonder if that's possible to shorten the number of end-to-end tests here without sacrificing test coverage. For example, if we already have unit tests that verify the same functionality, do we need to duplicate them with e2e tests? 

I mean, e2e tests are still valuable, as they help ensure that the system is functioning correctly and that all components are working together as intended. However, they are expensive in terms of maintenance. We can achieve a more balanced and effective testing strategy by supplementing unit tests with integrational tests. But duplication of test cases on different levels seems excessive to me.

Reply:
I'm afraid test coverage will suffer if we try to reduce the number of e2e tests. Unit tests rely on mocks and serve a bit different purpose. Even if we have supposedly duplicating unit and e2e tests in fact we have different environments they execute in. Unfortunately it's the price we have to pay to have better coverage.

I'm curious about @MadameSheema's opinion regarding this topic.